### PR TITLE
[App] fix panel requirements

### PR DESCRIPTION
**Pull request by adam-lightning**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by awaelchli — CHANGES_REQUESTED** [review]
Panel is an optional dependency. You can see [here](https://github.com/Lightning-AI/lightning/blob/208bf6faa87fca8d9015655f478c9048cba9840e/src/lightning_app/frontend/panel/panel_frontend.py#L32-L36) and [here](https://github.com/Lightning-AI/lightning/blob/208bf6faa87fca8d9015655f478c9048cba9840e/src/lightning_app/frontend/panel/panel_frontend.py#L84). It should not be added to the base requirements. This is in the same category as streamlit, for example.

**Comment by Borda** [comment]
for the context, this is failing on all PRs now:
https://github.com/Lightning-AI/lightning/runs/8092459965?check_suite_focus=true

File: requirements/app/base.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ s3fs>=2022.1.0, <=2022.7.1
 croniter # for now until we found something more robust.
 traitlets<5.2.0 # Traitlets 5.2.X fails: https://github.com/ipython/traitlets/issues/741
 arrow>=1.2.0, <=1.2.2
+panel>=0.12, <=0.13.1
```
Comment (Borda) [inline requirements/app/base.txt:10]:
```suggestion
```

**Comment by yurijmikhalevich** [comment]
@awaelchli, isn't it required to run the FE now?

**Comment by awaelchli** [comment]
> @awaelchli, isn't it required to run the FE now?

@yurijmikhalevich 
It is required if your app wants to use the Panel frontend. Otherwise not. Nothing in the framework requires panel as dependency, so that's why it is optional. If a user runs their app and they use the PanelFrontend API, but don't have panel installed, a friendly error will say "run pip install panel".

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**